Formula/cairo.rb

```diff
@@ -47,6 +54,10 @@ def install
       args << "--enable-xcb=no" << "--enable-xlib=no" << "--enable-xlib-xrender=no"
     end
 
+    if build.head?
+      system "./autogen.sh", *args
+    end
+
```

Comment:
They might be, some `autogen` scripts immediately throw themselves into `configure`. If that's the case you can usually opt-out via a variable or you'll need to `if/else` the `autogen` and `configure` steps.